### PR TITLE
varint: minor performance tweaks

### PR DESCRIPTION
**Pull request by samoht**
It's a bit hard to see the exact impact of these changes, but this seems to make a difference for large integers.

### CODE REVIEW

File: src/repr/type_binary.ml [hunk 1]
```diff
@@ -224,12 +225,12 @@ module Decode = struct
     (ofs, Int64.float_of_bits f)
 
   let int buf ofs =
-    let rec aux n p ofs =
+    let rec aux buf n p ofs =
```
Comment (pascutto) [inline src/repr/type_binary.ml:228]: Does passing `buf` to `aux` make a difference here?

**Review by pascutto — APPROVED** [review]

File: src/repr/type_binary.ml (hunk 1 above)
Comment (dinosaure) [reply src/repr/type_binary.ml:228]: I believe that it does not make a closure with an free-variable.

Comment (samoht) [reply src/repr/type_binary.ml:228]: yes I haven't checked the generated code but it should generate a loop if all the variable are bound as function parameters (might be worth checking)

Comment (pascutto) [reply src/repr/type_binary.ml:228]: Thanks for the clarification!

**Pull request merged**